### ISSUE TITLE
`inLiquidItemChancePoolDimBiome` not working - all fishing lead to default

### ISSUE THREAD

**Issue by Krutoy242** [ukmojb/CustomFishing#8]
There is my ZenScript code (some lines removed):

```ts
#modloaded customfishing

import crafttweaker.item.IItemStack;
import crafttweaker.liquid.ILiquidStack;
import mods.customfishing.FishingInCustomLiquid.inLiquidItemChancePoolDimBiome;

val fishing = {
  <fluid:pyrotheum>: {
    <mysticalagriculture:chunk:17>: 0.5,
    <ic2:dust:16>: 0.5,
    <mysticalagriculture:fire_essence>: 0.5,
    <thaumcraft:crystal_essence>.withTag({Aspects: [{amount: 1, key: "ignis"}]}): 0.5,
    <quark:soul_bead>: 0.5,
    <tconstruct:shard>.withTag({Material: "fierymetal"}): 0.2,
  },
} as float[IItemStack][ILiquidStack]$orderly;

for fluid, tuple in fishing {
  for item, chance in tuple {
    inLiquidItemChancePoolDimBiome(fluid, '', item, chance, '', null);
  }
}
```

Now when I fishing in Pyrotheum (from mod Thermal Expansion) I still get default fishes, same as if I would fish from Water (item instantly disappeared btw):

![javaw_jyMuI3Hm5D (1)](https://github.com/user-attachments/assets/0de734b4-3c5d-45ac-892c-d100495b2146)

Even default example, cited in mod description, not working.
This is my code:
```ts
mods.customfishing.FishingInCustomLiquid.inLiquidItemChancePoolDimBiome(<liquid:lava>, "", <minecraft:apple>, 0.9F, "", null);
```
As the result, fishing is happens extremely slow (1 fish per 60 seconds, even with `Lure III` ench), and fished items is match default fishing list (no apples).

**Comment by ukmojb** [comment]
Try 0.1.1,I think it's been fixed))))

**Comment by Krutoy242** [comment]
No, its not fixed. Now i cant fish anything, even in vanilla water.
There, i sit for several minutes and doesnt fish anything:

![mpc-hc64_KmkUuM5zkd](https://github.com/user-attachments/assets/98bce11b-904e-48b3-93fe-4f1ec31858d0)

**Comment by ukmojb** [comment]
Try 1.2 again (I changed again)

**Comment by ukmojb** [comment]
I am a senior high school student in China. I can only go home on weekends. If I can't do it again this time, I can only modify it again next week

**Comment by Krutoy242** [comment]
Why you cant just install E2EE modpack and try debug it yourself?

**Comment by ukmojb** [comment]
> Why you cant just install E2EE modpack and try debug it yourself?

Because I have to go to school))))))

**Comment by Krutoy242** [comment]
> Try 1.2 again (I changed again)

I tried - water tiles still cant be fished.

But now, modded liquids play fishing animation, but i cant pull hook out and no item fished out.


https://github.com/user-attachments/assets/37da7565-477a-49b8-a5de-1cc5f1e67395

**Comment by ukmojb** [comment]
https://github.com/user-attachments/assets/4982da13-6d77-4fd0-8d98-93ab4a4ecbad

[customfishing-0.1.2-test.zip](https://github.com/user-attachments/files/17965957/customfishing-0.1.2-test.zip)
I can use it in 1.48 version of e2ee, you can try to use it

**Comment by Krutoy242** [comment]
Ok, I will try it later. Will i need to wait another week for you to publish it if its working? 😅

**Comment by ukmojb** [comment]
If you can get back to me by 4 o 'clock in China time zone, no

**Comment by Krutoy242** [comment]
I still cant fish from fluid sources. Water give me nothing and in modded fluids i cant pull hook out:


https://github.com/user-attachments/assets/b85bef9e-126b-4186-9308-af65e194cb23

**Comment by Krutoy242** [comment]
Any updates?

**Comment by ukmojb** [comment]
> Any updates?

There's an update. You can check the github wiki. (The new version should have been fixed, probably.)

**Comment by ukmojb** [comment]
You can wait for a moment. I updated version 0.1.6

**Comment by ukmojb** [comment]
> Any updates?

Has it been solved?

**Comment by Krutoy242** [comment]
I will take a look in about a week or two.

**Comment by Krutoy242** [comment]
I did several tests with the [this script](https://github.com/Krutoy242/Enigmatica2Expert-Extended/blob/49933899deaf3e4fc7e4ab2fffeae808a8de0483/scripts/mods/customfishing.zs) and default config.

- ✅ Vanilla water fishing normal, default fishing loot table (mostly Harvestcraft fishes as usual).
- ❌ Any other liquid in game is fishable.
- ❌ Other liquids, `Pyrotheum` or `Molten Cheese` defined by scripts, have defaul Water fishing list.

<img width="479" height="301" alt="Image" src="https://github.com/user-attachments/assets/8aff5eb0-0aeb-4dd0-9e9c-17f6f5d7c3f4" />

[debug-customfishing.zip](https://github.com/user-attachments/files/23230129/debug-customfishing.zip)

**Comment by ukmojb** [comment]
I found that I actually made the effective code into comments. Please try this version again. And I have added more crt support, which can be viewed in https://github.com/ukmojb/CustomFishing/wiki

**Comment by ukmojb** [comment]
[CustomFishing-0.1.6-test.zip](https://github.com/user-attachments/files/23255153/CustomFishing-0.1.6-test.zip)

**Comment by Krutoy242** [comment]
**I'm tired** of testing your mod over and over again. It takes a lot of time, and the fact that the bug hasn't been fixed for almost a year makes me lose motivation to keep doing it again and again.

You have all the necessary information and ways to reproduce the bug. You can test your mod yourself by simply adding it to the modpack.

If you don't have time for that either, just leave the bug as it is until someone else wants to help you fix it.

**Comment by ukmojb** [comment]
I did solve the problem you mentioned during the test (just to confirm).
However, if you want to fish with any liquid, you need to enable the "Fishing in AnyLiquid" option in the configuration file

Last year, I was still a high school student in China and needed to take the college entrance examination, so I didn't have much time to fix the bugs
